Lesson20/selcan/oop.js

```diff
@@ -0,0 +1,106 @@
+// 1. Create a Person class with name, age, and gender properties, and a greet() method that logs a greeting message to the console. Then create a Student class that extends the Person class and adds a major property. Override the greet() method in the Student class to log a different greeting message to the console. Create 2 variables, one of type Person and one of type Student and log them to console.
+
+class Person {
+    constructor(name,age,gender){
+        this._name = name;
+        this._age = age;
+        this._gender = gender;
+    }
+    greet(){
+        return `Hi!My name is ${this._name}.I'm ${this._age} years old.I'm a ${this._gender}.`
+    }
+}
+class Student extends Person {
+    constructor(name,age,gender){
+        super(name,"yüzmeyi çok sever.",gender)
+    }
+    greet(){
+        return `I don't like swiming.`
+    }
+}
+const Student1 = new Student("Ali",23,"erkek");
+console.log(Student1.greet());
+console.log(Student1);
+const Person1 = new Person("jale",54,"kadin");
+console.log(Person1);
+
+
+
+// 2. Create a Shape class with name and color properties, and a describe() method that logs a description of the shape to the console (e.g. "This is a blue square"). Then create a Square class that extends the Shape class and adds sideLength and area properties. Override the describe() method in the Square class to log a description of the square to the console (e.g. "This is a blue square with a side length of 5 and an area of 25").
+class Shape {
+    constructor(name,color){
+        this._name = name;
+        this._color =color;
+    }
+    description(){
+        return `This is a ${this._color} square.`
+    }
+}
+class Square extends Shape{
+    constructor(name,color,sideLength){
+        super(name,color);
+        this._sideLength = sideLength;
+        this._area = sideLength * sideLength;
+    }
+    description(){
+        return `This is a ${this._color} square with a side length of ${this._sideLength} and an area of ${this._area}.`
+    }
+}
+const square1 = new Square("kare","blue",20)
+const Shape1 = new Shape("kare","pink");
```

Comment:
Use lowercase when naming normal variables
```
const shape1 = new Shape("kare","pink");
```

Reply:
Make sure to update console.log statements below that uses the 'shape1' after  changing the name here.